### PR TITLE
Buff Marine Armour During HvH

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

If the Faction Clash HvH game mode is enabled, this PR will buff all Marine armours to have significantly higher bullet, bomb and/or internal_damage (internal_damage reduces the chance of internal injuries like fractures) armour values. Some values are reduced in a few cases but buffed in others however. 

The explicit changes are as follows:

**Medium** (Includes Smartgunner, B12, IO)

- Bullet: 30 (from 20)
- Bomb: 20 (from 15)
- Internal: 30 (from 20)

**Light**

- Bullet: 20 (from 15)
- Bomb: 15 (from 20)
- Internal: 25 (from 10)

**Heavy**

- Bullet: 35 (from 35)
- Bomb: 35 (from 35)
- Internal: 30 (from 25)

**MP** (Includes CO, XO, SO armours)

- Bullet: 30 (from 20)
- Bomb: 25 (from 20)
- Internal: 30 (from 15)

Synthetic armour is unchanged in both game modes. 

Spec armours also get buffs similar to Medium armour, B18 gets a slight boost over normal. 

I have decided against effecting similar changes to Helmets, Gloves and Boots at this time. 

# Explain why it's good for the game

HvH is extremely lethal, especially so for Marines as their armour is not well suited to shrugging off bullets and bombs. 

Whilst this change does not significantly reduce the overall time to kill of Marines in combat, it does significantly reduce the occurrence rate of a marine suffering excessive fractures and internal bleeding from being gunned down. 

From testing, you should see less cases of a marine suffering three to fix fractures, and more cases of a marine suffering zero to two fractures when gunned down from full health with standard ammunition. 

Death remains punishing, but it will be far less oppressive once you are revived. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
balance: If the 'Faction Clash' human versus human game mode is enabled, all marine armours will have their defensive values significantly buffed to make marine armour far more valuable and effective in resisting permanent damage. The average Marine should experience less fractures and other permanent wounds from gunfire.  
/:cl:

### CODE REVIEW

**Comment by CABSHEP** [comment]
So Heavy armor trades speed and item slot for 5 additional bomb ressistance? Why would anyone pick Heavy Armor in HvH in that case?

Especially since half of the marine force and everyone besides CLF have a lot of AP with pen higher than 30.

I think Heavy armor should be massively reworked for HvH, slowing marine down even more, but given massive bullet and internal damage armor. Maybe even make it weaker against bombs in exchange, that would be interesting. 

B12, M4, IO armor and smartgunner mesh should also be buffed. Especially B12 which is 30 points purchase for being weaker than medium. 

With all of that AP everyone has access to this armor buff (30) still doesn't even break M39 AP.

**Comment by TheManWithNoHands** [comment]
Yeah, heavy armor just gettin extra bomb is kinda silly. Its more kevlar, it should protect you from more bullets. HvH is where heavy armor should shine.

Also, you migth need to touch some more armor then just those four. B12, Grenadier spec, CO, and the service jacket needs adjusted stats. Especially grenadier specs armor, that thing has set ALL armor values for itself.

**Comment by Zenith00000** [comment]
To make HvH not complete ass, please just disable IB (internal bleeding), literally the worst thing about it.

**Comment by Steelpoint** [comment]
@CABSHEP 
I'll add values to bring the Smartgunner's armour to the standard of Medium armour. I missed that.
B12, M4 and IO's armour get buffed to the value of Medium armour, plus the unique perks those armours get remain intact in HvH. I'm not as liable to buff B12 more since its the standard issue SL armour, and its already fairly competitive now. 
I agree with the Heavy Armour assessment and have returned its Bullet and Bomb values to normal. 

"With all of that AP everyone has access to this armor buff (30) still doesn't even break M39 AP."
AP ammo remains quite strong in HvH but, the occurrence rate of fractures is still decreased overall thanks to these buffs, though extreme overkill can still cause a lot of injuries but that's the nature of tick rates and not dying when you should be. 

@TheManWithNoHands 
CO armour falls under MP armour which has been buffed to be slightly better than Medium armour. 
Service jacket is not at all intended for combat use, if you really want maximum movement speed you pay the cost. It does share the same bullet defence as light armour but no bomb or internal protection, I don't think that needs to change.

@Zenith00000 
We should see less IB overall thanks to this change.
I could be convinced to add quick clot as a Corpsman item but, that would be a separate PR.

**Comment by TheManWithNoHands** [comment]
My guy, i told you it changes the bullet armor itself.
So it wont recive a buff at all. 
grafik
Although i do hope you still adress the Grenadier armor. Cause someone did an uppsy that overides your buffs.
grafik

**Comment by Steelpoint** [comment]
@TheManWithNoHands 
MP armour is buffed to 30 bullet defence, the CO already has 30 so, nothing changes as they have the intended bullet defence. 
Regarding the Specialists, I'm hoping to rework that in a separate PR so all the Specs are given the (fairly strong) B18 armour.

**Comment by Steelpoint** [comment]
I've also updated the Spec armours.

**Comment by Steelpoint** [comment]
If an issue with balance is constrained to how strong armour piercing ammo is, perhaps a separate PR can investigate in tweaking how much AP ammo both sides get access too, making it more rare? 

After doing testing I feel this is a good step for most guns.

File: code/modules/clothing/suits/marine_armor/_marine_armor.dm [hunk 1]
```diff
@@ -286,6 +286,17 @@
 	item_state_slots = list(WEAR_JACKET = "mp_armor")
 	black_market_value = 20
 
+/obj/item/clothing/suit/storage/marine/MP/Initialize(mapload)
+	. = ..()
+	if(SSticker.mode && MODE_HAS_FLAG(MODE_FACTION_CLASH))
+		armor_bullet = CLOTHING_ARMOR_HIGH
+		armor_bomb = CLOTHING_ARMOR_MEDIUMHIGH
+		armor_internaldamage = CLOTHING_ARMOR_HIGH
+	else
```
Comment (kiVts) [inline code/modules/clothing/suits/marine_armor/_marine_armor.dm:295]: You dont need this else and everything under it

File: code/modules/clothing/suits/marine_armor/_marine_armor.dm [hunk 2]
```diff
@@ -415,6 +426,17 @@
 	armor_variation = 0
 	specialty = "M3 pattern smooth marine"
 
+/obj/item/clothing/suit/storage/marine/medium/Initialize(mapload)
+	. = ..()
+	if(SSticker.mode && MODE_HAS_FLAG(MODE_FACTION_CLASH))
+		armor_bullet = CLOTHING_ARMOR_HIGH
+		armor_bomb = CLOTHING_ARMOR_MEDIUM
+		armor_internaldamage = CLOTHING_ARMOR_HIGH
+	else
```
Comment (kiVts) [inline code/modules/clothing/suits/marine_armor/_marine_armor.dm:435]: Generally every else you added is redundant, the stats are already set to be like that

File: code/modules/clothing/suits/marine_armor/_marine_armor.dm [hunk 3]
```diff
@@ -455,6 +477,17 @@
...
+/obj/item/clothing/suit/storage/marine/light/Initialize(mapload)
+	. = ..()
+	if(SSticker.mode && MODE_HAS_FLAG(MODE_FACTION_CLASH))
+		armor_bullet = CLOTHING_ARMOR_MEDIUM
+		armor_bomb = CLOTHING_ARMOR_MEDIUMLOW
+		armor_internaldamage = CLOTHING_ARMOR_MEDIUMHIGH
+	else
+		armor_bullet = CLOTHING_ARMOR_MEDIUMLOW
+		armor_bomb = CLOTHING_ARMOR_MEDIUM
+		armor_internaldamage = CLOTHING_ARMOR_LOW
```
Comment (kiVts) [inline code/modules/clothing/suits/marine_armor/_marine_armor.dm:489]:
```suggestion

```
Like that, the type is already defined with those stats, do that to every else you added

**Review by kiVts — DISMISSED** [review]
the else apocalypse

**Comment by Steelpoint** [comment]
Requested changes made.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by Steelpoint** [comment]
Made edits to conform to recent decorator changes

**Comment by Steelpoint** [comment]
Fixing merge conflict re-added something that was removed in a different PR, so I've fixed it.

**Review by harryob — APPROVED** [review]

**Pull request merged**